src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -1532,12 +1536,6 @@ internal SqlError ProcessSNIError(TdsParserStateObject stateObj)
                         //
                         errorMessage = SQL.GetSNIErrorMessage((int)details.sniErrorNumber);
 
-                        // If its a LocalDB error, then nativeError actually contains a LocalDB-specific error code, not a win32 error code
-                        if (details.sniErrorNumber == (int)SNINativeMethodWrapper.SniSpecialErrors.LocalDBErrorCode)
-                        {
-                            errorMessage += LocalDBAPI.GetLocalDBMessage((int)details.nativeError);
-                            win32ErrorCode = 0;
-                        }
```

Comment:
I believe it should throw a `PlatformNotSupportedException` on unsupported platforms. Is this by design or accidentally removed?

Reply:
Only connection string starting  with datasource (localddb) and (localdb)\ are considered localdb connection strings by SNI. MDS is sending named pipe connection string now so it wont trigger any localdb code in SNI so no localdb errors from SNI to parse are expected.